game-core/src/main/java/games/strategy/triplea/ui/screen/Tile.java

```diff
@@ -38,19 +35,19 @@ public boolean needsRedraw() {
     return isDirty && !isDrawing;
   }
 
-  public void acquireLock() {
-    LockUtil.INSTANCE.acquireLock(lock);
-  }
-
-  public void releaseLock() {
-    LockUtil.INSTANCE.releaseLock(lock);
+  /**
+   * Because our Tile-rendering system doesn't use something like "double buffers" on a higher level
+   * (the swing components support it internally, but we don't draw in batch) we expose this
+   * internal mutex to prevent tiles being drawn mid-refresh.
+   */
+  public Object getMutex() {
+    return mutex;
   }
 
   /** Returns the image representing this tile, re-rendering it first if the tile is dirty. */
   public Image getImage(final GameData data, final MapData mapData) {
     if (isDirty) {
-      acquireLock();
-      try {
+      synchronized (mutex) {
```

Comment:
Not in this case, on a semantic level the code here should be equivalent,  so no reason to check performance.

Reply:
Are you sure? The read locks were global before, in this case the mutex is specific to just Tile.java. For example, if there is a read lock on game data from a different class, before that would not have blocked this lock. Now that we are have a single-class mutex, presumably it would.

Reply:
Hmm I could be missing something, but previously we used a re-entrant lock in this class, and now we use the synchronised keyword so it should be the same.